SignalServiceKit/src/Messages/OWSMessageManager.m

```diff
@@ -278,6 +278,7 @@ - (void)handleEnvelope:(OWSSignalServiceProtosEnvelope *)envelope
             [self handleIncomingEnvelope:envelope withSyncMessage:content.syncMessage transaction:transaction];
 
             [[OWSDeviceManager sharedManager] setHasReceivedSyncMessage];
+            [OWSDeviceManager.sharedManager setMayHaveLinkedDevices:YES transaction:transaction];
```

Comment:
In the interest of caution, I'm also setting this flag whenever we receive a sync message.

Outcome:

Phone will _erroneously_ not send sync message IFF all of the following are true:

* Phone previously (with an older version) added a linked device but never sent a sync message (this should be extremely rare).
* The linked device never sends a sync message to the phone.

That should be extremely rare - and we should recover as soon as the user does anything significant on linked device.